proposed/2022/pack-referenced-projects.md

```diff
@@ -0,0 +1,71 @@
+Chaining together dozens of arguments to achieve parity with metadata usually automagically introduced by the SDK becomes tedious, which eventually weakens the otherwise rich set of information attached to a modern NuGet package.
+
+## Explanation
+
+### Functional explanation
+
+Proposed solution is to add new `Pack` and `PackagePath` properties to `ProjectReference` (`<ProjectReference Include="..." Pack="True" PackagePath="tools" />`) and `PackageReference`. 
+If `Pack` is `False` or not present - consider the project (or package) as a NuGet's dependency, if it's `True` - pack its artifacts in.
+If `PackagePath` is not present - pack artifacts to the default location within the package. If it's present - pack them to the specified path.
+
+For `PackageReference`s it's useful in case referenced package is from private feed, but NuGet that's being packed will be public.
+
+### Technical explanation
+
+When NuGet is being created from `csproj` file:
+
+1) If `ProjectReference` doesn't have `Pack` property, or has it set to `False` - preserve behavior that have existed before introduction of this feature (don't include artifacts of referenced projects into the NuGet, and consider them as NuGet-dependencies).
+2) If `ProjectReference` has `Pack` property and it's set to `True` - include it's artifacts into the NuGet (to the default location or one specified in `PackagePath`).
+3) Do the same steps for `PackageReference`s
+
+## Drawbacks
+
+None
+
+## Rationale and alternatives
+
+- Use `IsPackable` property from referenced project to determine it it's supposed to be packed or be a NuGet dependency.
+
+It will be more automated, but won't fit some edge cases (also it will be a breaking change). And it isn't applicable to `PackageReferences`.
+
+- Introduce new project-wide `PackReferencedProjects` property that would switch pack behavior for all `ProjectReference`s inside a single project.
+
+This one is less flexible since it cannot switch on per-`ProjectReference` basis. And it isn't applicable to `PackageReferences`.
+
+- Another solution is to introduce a `IncludeReferencedProjects` switch for `nuget` CLI.
+
+Downsides are: Behavior of individual projects (if packing is performed in bulk) and `ProjectReference`s cannot be changed, and it's not obvious how to properly pack any specific project (I thought we are consolidating all the information in `csproj` now). And it isn't applicable to `PackageReferences`.
+
+## Prior Art
+
+This design flaw have been a pain for the community for more than 5 years now. It was such an issue that even third party solution was created - [nugetizer](https://github.com/devlooped/nugetizer).
+
+I haven't used it, so don't have much to comment here, but community clearly [expressed a desire for out of the box solution](https://github.com/NuGet/Home/issues/3891#issuecomment-1080044314).
+
+## Unresolved Questions
+
+- **How do we flow/assign TFM based `lib`/`content` items?** (Is `PackagePath` a desired parameter?)
```

Comment:
I'm not sure I fully understand this example. Can you clarify better what the ask is?

Reply:
Referring to the conversation earlier - here: https://github.com/NuGet/Home/pull/11708#discussion_r844459807

Reply:
Thanks for that link. In that case, I think the compat side would probably need to not be configurable. 

In other words, given a project reference today, the particular assembly chosen by NuGet/SDK is the best compatible framework one, so I think the switch would only really need to be copy for project, or don't. 

The conflict resolution between package items and references is more complex as that happens in a part that NuGet doesn't understand. So we'd to NuGet and the SDK to work through that in some way.

Reply:
I understand there is *potentially* a better framework for some of the dependent assemblies, but accounting for that would make things too complicated imo. NuGet today chooses a full lib folder, so we can't really pick and choose from different folders.